### PR TITLE
Fix endDecorator color for `ControlBase`

### PR DESCRIPTION
**Pull request by toha993**

### CODE REVIEW

File: packages/react/src/control-base/ControlBase.tsx [hunk 1]
```diff
@@ -29,7 +29,6 @@ export const ControlBase = forwardRef<HTMLDivElement, ControlBaseProps>(
   ) => {
     const { restProps, sprinkleProps } = extractSprinkles(props);
     const id = useId(idProp);
-
```
Comment (radnan) [inline packages/react/src/control-base/ControlBase.tsx:32]: Please keep the bank line here

We should always use blank lines to group related blocks of code - kinda like paragraphs in text

File: packages/react/src/control-base/ControlBase.tsx [hunk 2]
```diff
@@ -44,7 +43,11 @@ export const ControlBase = forwardRef<HTMLDivElement, ControlBaseProps>(
         </Text>
 
         {endDecorator && (
-          <Box asChild {...styles.decorator()}>
+          <Box
+            asChild
+            color={props.disabled ? "fg.disabled" : "fg.secondary"}
```
Comment (radnan) [inline packages/react/src/control-base/ControlBase.tsx:48]: Let's add the fontSize here as well

In majority cases I'm assuming we'll need small text

File: packages/react/src/control-base/ControlBase.tsx (hunk 1 above)
Comment (toha993) [reply packages/react/src/control-base/ControlBase.tsx:32]: My fault vai, I added a log in this line, later removed the log and it happened.

File: apps/storybook/src/forms/Checkbox.stories.tsx [hunk 3]
```diff
@@ -15,11 +15,7 @@ export const Basic: Story = {};
 
 export const HelperText: Story = {
   args: {
-    endDecorator: (
-      <Text color="fg.secondary" fontSize="sm">
-        Helper Text
-      </Text>
-    ),
+    endDecorator: <Text>Helper Text</Text>,
```
Comment (radnan) [inline apps/storybook/src/forms/Checkbox.stories.tsx:18]: technically we can just change these to strings
```suggestion
    endDecorator: "Helper Text",
```

File: apps/storybook/src/forms/RadioGroup.stories.tsx [hunk 4]
```diff
@@ -125,7 +107,7 @@ export const ComplexExample1: Story = {
         <Flex className={styles.item} {...props} gap="xs">
           <RadioGroupItem
             endDecorator={
-              <Text color="fg.secondary" fontSize="sm">
+              <Text color="fg.secondary">
                 Displays a message as a bar at top of page
               </Text>
```
Comment (radnan) [inline apps/storybook/src/forms/RadioGroup.stories.tsx:112]: let's change these to strings as well
```suggestion
              "Displays a message as a bar at top of page"
```

File: packages/react/src/control-base/ControlBase.tsx (hunk 1 above)
Comment (radnan) [reply packages/react/src/control-base/ControlBase.tsx:32]: can you bring back the blank line in that case please

Comment (radnan) [reply packages/react/src/control-base/ControlBase.tsx:32]: also needs a rebase with `main`

**Review by radnan — APPROVED** [review]

**Pull request merged**